lib/src/commit.rs

```diff
@@ -80,11 +80,11 @@ impl Commit {
         &self.data.parents
     }
 
-    pub fn parents(&self) -> Vec<Commit> {
+    pub fn parents(&self) -> BackendResult<Vec<Commit>> {
```

Comment:
fwiw, I changed `op.parents()` to iterator of `Result<Operation>` because it was easier to adapt to dag_walk functions.

Reply:
I noticed that after sending this PR. I wasn't sure if we should do the same for `Commit::parents()` because it's called quite frequently. We could have a separate `parents_iter()` for when you want an iterator over `BackendResult` instead. What do you think?

Reply:
I don't have a strong opinion, but `.predecessors()` will be probably better to return `Iterator<Item = Result<..>>`, so `commit.parents()` will be an outlier.

Maybe we can also add a generic (or specialized) version of `merge_commit_trees(repo, &commit.parents())`? Appears that many `.parents()` callers just need a merged parent tree.

Reply:
I sent https://github.com/martinvonz/jj/pull/3645 to add the helper for merging the parents of a commit, or at least I think that's what you meant

Reply:
I've updated this the functions to return iterators now.